tests/unit/_experimental/dataset/test_common.py

```diff
@@ -72,9 +70,12 @@ def test__validate_dataframe_ids(df: pd.DataFrame, id_fields: List[str]) -> None
     [
         # dataframe is missing one of id_fields
         (pd.DataFrame(dict(a=[1, 2, 3])), ["a", "b"]),
+        # dataframe id_fields datatypes are not primitives
+        (pd.DataFrame(dict(a=[dict(c=i) for i in range(3)], b=[1, 2, 1])), ["a", "b"]),
+        (pd.DataFrame(dict(a=[[1], [2], [3]], b=[1, 2, 1])), ["a", "b"]),
         # dataframe values in id_fields is not unique
         (pd.DataFrame(dict(a=[1, 2, 1], b=[1, 2, 1])), ["a", "b"]),
-        (pd.DataFrame(dict(a=[[1], [1], [1]], b=[1, 2, 1])), ["a", "b"]),
+        (pd.DataFrame(dict(a=[1, 1, 1], b=[1, 2, 1])), ["a", "b"]),
         # the key sequence difference will not make it unique
```

Comment:
this comment is obsolete, maybe we should remove all tests under this comment as well?